### ISSUE TITLE
auxlib build appears broken in SN 0.5.0 on Scala 2.12.13.

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#3265]
When I try to build Scala Native 0.5.0-SNAPSHOT itself on Scala 2.12.13 & 2.12.14, I get three instances of:

```
/auxlib/src/main/scala/scala/collection/concurrent/CNodeBase.scala:15:18: type arguments [_$1,_$2] do not conform to class CNodeBase's type parameter bounds [K <: AnyRef,V <: AnyRef]
[error]       fromRawPtr(classFieldRawPtr(obj, "csize"))
```

The build succeeds with Scala 2.12.17.  I did not try the intermediate versions.

I understand and support that not all Scala versions are tested in CI, particularly
older versions.  I appreciate the reduced wall-clock time waiting to see if a PR I submit
passes CI on version I do care about.

Understood that this is `auxlib` and not something shipped to the World, but it still seems like
flies in the dogfood.

I got to this pain point because I was verifying a PR I hope to submit before the snow flies on
both the lowest and highest version of Scala supported by Scala Native.

**Comment by ekrich** [comment]
I am agreeing with @armanbilge on this that we shouldn't try to support any previous patch releases for 2.12 and 2.13 especially if the fixes make Scala Native work. I think on Scala 3, it would be good to support 3.1.x plus or whatever that version is but if we have a big need for a newer version then I think we have to drop the backward compat to whatever the new minimum 3.x version would be.

**Comment by armanbilge** [comment]
Sorry, let me clarify: my point is that in order to support Scala 2.12.13, we don't need to be able to compile auxlib on Scala 2.12.13.

We _do_ need to be able to compile (1) the nsc plugin and (2) the junit plugin:
1. https://repo1.maven.org/maven2/org/scala-native/nscplugin_2.12.13/
2. https://repo1.maven.org/maven2/org/scala-native/junit-plugin_2.12.13/

Everything else can be compiled with any Scala 2.12 version, because Scala 2.12.x promises forwards and backwards compatibility in that series. For example, nativelib is just published for 2.12, not a specific patch version:

https://repo1.maven.org/maven2/org/scala-native/nativelib_native0.4_2.12/

In summary: so long as the latest compiler plugin is published for a specific Scala version, we have not dropped support for that Scala version.

**Comment by ekrich** [comment]
I guess one final point is that if someone uses a older point release of 2.12 is this case, the only remedy is to upgrade to the latest maintenance release. This seems to be some sort of local change and/or change in the older compiler that together that are having problems. I could be wrong be I would say you shouldn't worry about it.

**Comment by LeeTibbert** [comment]
@arman, @ekrich 

Thank you both for your comments.  

I still think this is a defect. Given the info & perspective from the two of you, I will push
it to the back of my priority list and limit the low end of my pre-PR verification to SN 2.12.17.

In a private project "known errors" have a purpose.  In an OpenSource project,
trying to encourage PRs, they lead to people wasting time and getting a bad taste.

In this case, I suspect the breakage was not recognized at the time, hence not "known". 
I think it is probably unintentional.  

If the intention was that it is "acceptable" due to closely held release process lore, then I
will try to put some covers over the open cesspool with the hope of saving other people
the long day & night I wasted on this.

**Comment by LeeTibbert** [comment]
Added a Contributor's Note note in PR #3260 which sums up, in polite words the essence of
the discussion in this Issue: Just Don't Do That!

My hope is that Note will save some future devo, perhaps me, from falling into that pothole.
